src/lib/components/Button/Button.svelte

```diff
@@ -0,0 +1,44 @@
+<script lang="ts">
+  import "./Button.scss";
+  import classNames from "$lib/util/classNames";
+  import type { Variant, Type } from "./buttonOptions";
+  import warn from "$lib/util/warn";
+
+  export let disabled = false;
+
+  export let unstyled = false;
+
+  export let variant: Variant = "primary";
+
+  const variantClassesDict: Record<Variant, string[]> = {
+    primary: [],
+    base: ["usa-button--base"],
+    inverse: ["usa-button--inverse"],
+    "text-only": ["usa-button--text-only"],
+    outline: ["usa-button--outline"],
+    "outline-inverse": ["usa-button--outline-inverse"],
+    big: ["usa-button--big"],
+    "big-inverse": ["usa-button--big-inverse"],
+  };
+
+  $: variantClasses = unstyled ? [] : variantClassesDict[variant];
+
+  $: if (unstyled && variant !== "primary") {
+    warn("variant has no effect on an unstyled button!");
+  }
+
+  export let type: Type = "button";
+
+  let className = "";
+  export { className as class };
+</script>
+
+<button
+  {type}
+  {disabled}
+  aria-disabled={disabled}
+  class={classNames("usa-button", unstyled && "usa-button--unstyled", ...variantClasses, className)}
```

Comment:
Might be better to move the result of `classNames()` into a reactive value; I'm seeing some oddities in Storybook when switching around the values of different props (can't repro anything consistently unfortunately, and I'm seeming to have different results between running it locally versus building and serving it).

I think it would also be useful to add a test around ensuring that the expected classes get applied given different variants.